### DISCUSSION TITLE
resetExcept(<array of all properteies>)  calls reset() and clears all properties

### DISCUSSION THREAD

**Discussion by misenhei** [livewire/livewire#8096] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.8

### Laravel version

v10.47

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

In livewire/livewire/src/Concerns/InteractsWithProperties.php function resetExcept(), if  $properties == $this->all than  $keysToReset is an empty array and ends up calling $this->reset([]) reseting all properties. I admit it is odd to do, but I expected it to not reset the properties provided. 

current: livewire/livewire/src/Concerns/InteractsWithProperties.php
```
trait InteractsWithProperties
{
      ...
      protected function resetExcept(...$properties)
      {
          if (count($properties) && is_array($properties[0])) {
              $properties = $properties[0];
          }
      
         $keysToReset = array_diff(array_keys($this->all()), $properties);
      
         $this->reset($keysToReset);
      }
    ...
```

suggested: livewire/livewire/src/Concerns/InteractsWithProperties.php
```
trait InteractsWithProperties
{
      ...
      protected function resetExcept(...$properties)
      {
          if (count($properties) && is_array($properties[0])) {
              $properties = $properties[0];
          }
      
         $keysToReset = array_diff(array_keys($this->all()), $properties);
         
         if (!empty($keysToReset) {
           $this->reset($keysToReset);
         }
      }
    ...
```

### Repository or Wirebox URL

/

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@misenhei thanks for reporting! PRs to fix are welcome 🙂

**Comment by liamduckett** [comment, marked as answer]
Hey, this has been fixed in https://github.com/livewire/livewire/pull/9117 and released in [v3.6.3](https://github.com/livewire/livewire/releases/tag/v3.6.3)
